fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/SynchronizedSessionTest.java

```diff
@@ -245,33 +246,39 @@ private void testRenewLeaseContinuously(boolean reuseRunner) throws Exception {
                         e.printStackTrace();
                     }
                     SynchronizedSessionRunner runner = reuseRunner ? session1Runner0 : database.newRunner()
-                            .toSynchronized(lockSubspace, session1, 1_000);
+                            .joinSynchronizedSession(lockSubspace1, session1, 1_000);
                     checkActive(runner);
                 }
                 session1Stopped.set(true);
             });
             Thread tryStartSession = new Thread(() -> {
                 while (!session1Stopped.get()) {
-                    assertFailedStartSession(lockSubspace);
+                    assertFailedStartSession(lockSubspace1);
                     try {
                         Thread.sleep(random.nextInt(500));
                     } catch (InterruptedException e) {
                         e.printStackTrace();
                     }
                 }
             });
+
+            ConcurrentLinkedQueue<Throwable> exceptions = new ConcurrentLinkedQueue<>();
+            Thread.setDefaultUncaughtExceptionHandler((thread, exception) -> exceptions.add(exception));
+
             longSession.start();
             tryStartSession.start();
             tryStartSession.join();
+
+            assertTrue(exceptions.isEmpty());
```

Comment:
Is there a way to compare this to an empty collection or something so that if it fails, we can see what the exceptions were? (And in the other place where exceptions are added to a queue.)

Reply:
Added logging exceptions.